source/user/dft.rst

```diff
@@ -6,495 +6,145 @@ Density functional theory (DFT)
+
+Variationally minimizing the total energy with respect to the density yields the KS equations for obtaining the non-interacting reference orbitals, on par with HF theory, and these have the same general form as the Fock equations in :numref:`theory_scf`. However, the exact exchange, :math:`\hat{K}`, is replaced by the *xc* potential, :math:`\hat{v}_{\rm xc}=\delta E_{\rm xc}/\delta \rho`. For hybrid and meta-GGA calculations, PySCF uses the generalized KS formalism :cite:`GKS`, in which the so-called generalized KS equations minimize the total energy with respect to the orbitals themselves. 
+
+.. _user_dft_predef_func:
+
+Predefined *xc* Functionals and Functional Aliases
+==================================================
+
+The choice of *xc* functional is assigned via the attribute :attr:`DFT.xc`. This is a comma separated string (precise grammar discussed :ref:`below <user_dft_custom_func>`, e.g., ``xc = 'pbe,pbe'`` denotes PBE exchange plus PBE correlation. In common usage, a single name (alias) is often used to refer to the combination of a particular exchange and correlation approximation instead. To support this, PySCF will first examine a lookup table to see if :attr:`DFT.xc` corresponds to a common compound name, and if so, the implementation dispatches to the appropriate exchange and correlation forms, e.g., ``xc = 'pbe'`` directly translates to ``xc = 'pbe,pbe'``. However, if the name is not found in the compound functional table, and only a single string is given, it will be treated as an exchange functional only, e.g., ``xc = 'b86'`` leads to B86 exchange only (without correlation). Please note that earlier PySCF versions (1.5.0 or earlier)
+did not support compound functional aliases, and both exchange and correlation always had to be explicitly assigned. 
+
+PySCF supports two independent libraries of *xc* functional implementations, namely `Libxc <https://www.tddft.org/programs/libxc/>`_ and `XCFun <https://xcfun.readthedocs.io/en/latest/>`_. The former of these is the default, but the latter may be chosen upon by setting ``DFT._numint.libxc = dft.xcfun``, cf. `dft/32-xcfun_as_default.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/32-xcfun_as_default.py>`_. For complete lists of available functionals, the user is referred to `pyscf/dft/libxc.py <https://github.com/pyscf/pyscf/blob/master/pyscf/dft/libxc.py>`_ and `pyscf/dft/xcfun.py <https://github.com/pyscf/pyscf/blob/master/pyscf/dft/xcfun.py>`_, respectively. For instance, the two libraries may be switched between in order to leverage features that are exclusive to one of them. The example on `dft/12-camb3lyp.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/12-camb3lyp.py>`_ showcases one such use case for range-separated *xc* functionals (e.g., CAM-B3LYP), as Libxc only supports the energy and nuclear gradients for this functional, but not nuclear Hessians and TD-DFT gradients, for which the XCFun library is needed.
```

Comment:
Omitting the list of functionals from the documentation is a good idea, since it becomes quickly outdated. The updated Libxc binding also allows querying Libxc directly for the list of supported functionals, which can be triggered by a config keyword. If users compile PySCF against the libxc master branch, PySCF can be used with functionals that are not yet included in a point release of Libxc.

Reply:
One could consider adding an example that prints out the list of functionals in the current version of PySCF, i.e. the `XC_CODES` array in `libxc.py` and `xcfun.py`

Reply:
Agreed. Please see commit e451415

Reply:
OK